src/lib/SlaveApiClient.js

```diff
@@ -17,31 +17,57 @@
 
 "use strict";
 
-let xmlrpc = require('xmlrpc');
+let xmlrpc = require('@sixriver/xmlrpc');
 
 //-----------------------------------------------------------------------
 
-class SlaveApiClient {
+class AbortedError extends Error {
+    constructor() {
+        super();
+
+        Object.setPrototypeOf(this, AbortedError.prototype);
+    }
+}
 
+class SlaveApiClient {
+  
   constructor(host, port) {
     this._xmlrpcClient = xmlrpc.createClient({host: host, port: port});
+    this.requests = new Set();
   };
 
   requestTopic(callerId, topic, protocols) {
     let data = [callerId, topic, protocols];
     return new Promise((resolve, reject) => {
-      this._xmlrpcClient.methodCall('requestTopic', data, (err, resp) => {
-        if (err || resp[0] !== 1) {
+      let request = this._xmlrpcClient.methodCall('requestTopic', data, (err, resp) => {
+        this.requests.delete(request);
+        if(err instanceof AbortedError) {
+          reject(err);
+        }
```

Comment:
Is this `if` statement actually needed?  The `else if` rejects with `err` if `err` is defined.  It does ad `resp`, but shouldn't that be ok to add even if `err` is an AbortedError since the thing that handles AbortedError ignores `resp`.

Reply:
that's a good point. it was useful for testing so i could throw a breakpoint in there to make sure the type was coming through. i'll back that change out